bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -176,9 +176,6 @@ pub fn s2n_tls_pair(config: crate::raw::config::Config) {
     server
         .set_config(config.clone())
         .expect("Failed to bind config to server connection");
-    server
-        .set_client_auth_type(s2n_tls_sys::s2n_cert_auth_type::NONE)
-        .expect("Unable to set server client auth type");
```

Comment:
why is this not needed anymore?

Reply:
"None" is the default. You only need to call the method if you want to change it.

We could theoretically be [turning client auth on via the config](https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L1934) and need to turn it back off via the connection, but that's not supported by the bindings yet. Even if it was supported by the bindings, we certainly shouldn't be silently overriding random config settings in this method. Developers would definitely not expect that.

Reply:
My bad, setting the client auth type on the config instead of the connection is actually [currently supported by the bindings](https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls/src/raw/config.rs#L259-L268). But we're not calling it anywhere in that testing module.